core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -335,4 +335,26 @@ package object config {
         "spark.")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val SHUFFLE_FILE_BUFFER_SIZE =
+    ConfigBuilder("spark.shuffle.file.buffer")
+      .doc("Used to set the buffer size of outputstream in shuffle writer.")
+      .bytesConf(ByteUnit.KiB)
+      .checkValues(Set(4, 8, 16, 32, 64, 128, 256, 512, 1024))
```

Comment:
I have modify it.
For buffer size, I think 0 is meaningless, 
so I have modify for
`checkValue(v => v > 0 && v <= Int.MaxValue / 1024, ...)`
do you agree it?